src/qt/sendcoinsdialog.cpp

```diff
@@ -397,7 +406,11 @@ void SendCoinsDialog::sendButtonClicked([[maybe_unused]] bool checked)
     const QString confirmation = tr("Confirm send coins");
     const QString confirmButtonText = tr("Send");
     const QString psbt_button_text = tr("Create Unsigned");
-    auto confirmationDialog = new SendConfirmationDialog(confirmation, question_string, informative_text, detailed_text, SEND_CONFIRM_DELAY, confirmButtonText, !model->wallet().privateKeysDisabled(), psbt_button_text, this);
+
+    // Show the Create Unsigned button if set in the options or if private keys are disabled.
+    bool show_unsigned_button = model->getOptionsModel()->getEnablePSBTControls() || model->wallet().privateKeysDisabled();
```

Comment:
In commit "qt: hide Create Unsigned button behind an expert mode option" (790733e54c3daff547acbc90fb179b48b6d2fb3c)

It seems clumsy for this show unsigned logic to have to be repeated two places (in spending and fee bumping) and for all these extra calls to privateKeysDisabled to be needed in the code. It would be clearer to replace the `show_unsigned` constructor argument with an `always_show_unsigned` constructor argument and pass `getEnablePSBTControls` directly instead of `getEnablePSBTControls || privateKeysDisabled`, since SendConfirmationDialog already knows the `privateKeysDisabled` value

Also it would make the PR clearer if instead of changing the `SendConfirmationDialog` gradually in two commits, you just implemented it fully in the first commit passing `true` for `always_show_unsigned`, and then simply replaced `true` with `getEnablePSBTControls` in the second commit.